sphinx/api/configuration/file-creation.rst

```diff
@@ -25,6 +25,42 @@ root directory to install file targets with a relative path:
       :opt svn: The Subversion scheme. The location is a Subversion URL or
          an FCM location keyword. A URI with these schemes ``svn``,
          ``svn+ssh`` and ``fcm`` are automatically recognised.
+      :opt git: The Git scheme. The location is complex due to Git semantics.
+         It must have the scheme ``git`` and be of the form
+         ``git:REPOSITORY_URL::PATHSPEC::TREEISH``. ``REPOSITORY_URL`` should
+         be a Git repository URI which may itself have a scheme ``ssh``,
+         ``git``, ``https``, or be of the form ``HOST:PATH``, or ``PATH`` for
+         local repositories. ``PATHSPEC`` should be a path to a file or
+         directory that you want to extract. ``TREEISH`` should be a tag,
+         branch, or long commit hash to specify the commit at which you want
+         to extract. These should follow the same semantics as if you git
+         cloned ``REPOSITORY_URL``, git checkout'ed ``TREEISH``, and extracted
+         the path ``PATHSPEC`` within the clone. ``PATHSPEC`` must end with a
+         trailing slash (``/``) if it is a directory. To extract from the root
+         of the repository use a ``PATHSPEC`` of ``./`` e.g.
+         ``git:git@github.com:metomi/rose::./::2.2.0``. It may help to think
+         of the parts of the location as git:Where::What::When. Examples:
+
+         .. code-block:: rose
+
+           # Download the sphinx directory from the master branch of
+           # the github.com/metomi/rose repo. Note trailing slash.
+           [file:rose-docs]
+           source=git:git@github.com:metomi/rose::sphinx/::master
+
+           # Extract the whole contents of version 2.0.1 of the local
+           # repository at /home/user/some/path/to/my/git/repo.
+           [file:all_of_my_repo]
+           source=git:/home/user/some/path/to/my/git/repo::./::2.0.1
+
+           # Extract a single file from a particular commit of a repo
+           # on a machine that we have ssh access to.
+           [file:my_file]
+           source=git:machine01:/data/user/my_repo_name::etc/my_file::7261bff4d9a6c582ec759ef52c46dd794fe8794e
```

Comment:
I've tested these examples (substituting suitable dirs/hosts), and they all appear to work.

However, I get `[WARN] command not defined` in the output:

```console
$ rose app-run
...
[INFO] create: all_of_my_repo
[INFO] install: all_of_my_repo
[INFO]     source: remote:/home/h06/osanders/rose ref:2.0.1 commit:47e8c4edf843c776ab087d0c50020fafd59c0eca path:./ (git:/home/h06/osanders/rose::./::2.0.1)
[INFO] create: rose-docs
[INFO] install: rose-docs
[INFO]     source: remote:git@github.com:metomi/rose ref:master commit:3d28ca2cf2869de8b7cd49b3aaea8900275512cd path:sphinx/ (git:git@github.com:metomi/rose::sphinx/::master)
[INFO] install: my_file
[INFO]     source: remote:vld601:/home/h06/osanders/rose ref:3d28ca2cf2869de8b7cd49b3aaea8900275512cd commit:3d28ca2cf2869de8b7cd49b3aaea8900275512cd path:CHANGES.md (git:vld601:/home/h06/osanders/rose::CHANGES.md::3d28ca2cf2869de8b7cd49b3aaea8900275512cd)
[WARN] command not defined


```